### PR TITLE
Added NVMe support

### PR DESCRIPTION
**Pull request by turtle-fly**
For NVMe option `cmb_size_mb`, we customize test to be run only
on qemu version `>=2.10`. This feature is implemented by a
decorator in infrasim.helper.

Move concrete behavior and attribute of IDEDrvie and SCSIDrive
to their derived class, instead of being in CBaseDrive. Based on
this, we provide NVMeController as a derived class from
CBaseDrive.

**kind of weird**...but we decided to make NVMeController inherits
CBaseDrive instead of CBaseStorageController, due to it needs a
drive backend.

Signed-off-by: bcyan <448162474@qq.com>

### CODE REVIEW

**Comment by emcinfrasim** [comment]
```
******* Unit Test ******* 
TestName: test_set_nvme_controller_default_serial 
ErrorType: exceptions.AssertionError 
ErrorMessage:  
```

**Comment by emcinfrasim** [comment]
```
******* Functional Test ******* 
TestName: test_ipmi_console_stops_followed_by_node_destroy 
ErrorType: exceptions.AssertionError 
ErrorMessage: 
-------------------- >> begin captured stdout << ---------------------
[ 18739  ] test-socat starts to run
[ 18742  ] test-bmc starts to run
[ 18751  ] test-node is running
[ 18764  ] test-racadm starts to run
[ 18764  ] test-racadm stop
[ 18751  ] test-node stop
[ 18742  ] test-bmc stop
[ 18739  ] test-socat stop
Node test runtime workspace is destroyed.

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
1Ipmi-console: INFO: SIGTERM is sent to pid: 18529
1Model: INFO: [compute] infrasim can't find numactl in this environment
1Model: INFO: [ 18524  ] test-racadm stop
1Model: INFO: [ 18513  ] test-node stop
1Model: INFO: [ 18504  ] test-bmc stop
1Model: INFO: [ 18501  ] test-socat stop
1Model: INFO: [compute] infrasim can't find numactl in this environment
1Model: INFO: [BaseDrive] Creating drive: /home/infrasim/.infrasim/test/disk-sata-0-0.img
1Model: INFO: test-socat's command line: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test/.pty0,waitslave unix-listen:/home/infrasim/.infrasim/test/.serial,fork
1Model: INFO: [ 18739  ] test-socat starts to run
1Model: INFO: test-bmc's command line: ipmi_sim -c /home/infrasim/.infrasim/test/etc/vbmc.conf -f /home/infrasim/.infrasim/test/data/dell_r730.emu -n -s /home/infrasim/.infrasim/test/data
1Model: INFO: [ 18742  ] test-bmc starts to run
1Model: INFO: [ 18751  ] test-node is running
1Model: INFO: test-racadm's command line: racadmsim test 0.0.0.0 10022 admin admin auto
1Model: INFO: [ 18764  ] test-racadm starts to run
1Model: INFO: [ 18764  ] test-racadm stop
1Model: INFO: [ 18751  ] test-node stop
1Model: INFO: [ 18742  ] test-bmc stop
1Model: INFO: [ 18739  ] test-socat stop
1Model: INFO: [Node] Node test runtime workspcace is destroyed
--------------------- >> end captured logging << --------------------- 
```

File: test/functional/test_configuration_change.py [hunk 1]
```diff
@@ -185,6 +186,32 @@ def test_qemu_boot_from_disk_img(self):
...
         ssh.close()
 
+    @helper.qemu_version(">=2.10")
+    def test_auto_add_nvme_serial(self):
+        self.conf["compute"]["storage_backend"] = [{
+            "type": "nvme",
+            "cmb_size": 256,
+            "drives": [{"size": 8}]
+        }]
+        self.conf["compute"]["qemu_bin"] = "/home/infrasim/isilon/idic/qemu/x86_64-softmmu/qemu-system-x86_64"
```
Comment (fub2) [inline test/functional/test_configuration_change.py]: This is temporary solution to hard-code where compatible qemu binary is placed. May be add a comment of "fixme".

Comment (turtle-fly) [reply test/functional/test_configuration_change.py]: Good catch! This config is a temporary local test config, to verify a good behavior on qemu 2.10, which should be removed.

**Pull request merged**